projects/igniteui-angular/src/lib/grids/hierarchical-grid/row-island.component.ts

```diff
@@ -90,15 +91,51 @@ export class IgxRowIslandComponent extends IgxHierarchicalGridBaseDirective
     @ContentChildren(IgxColumnComponent, { read: IgxColumnComponent, descendants: false })
     public childColumns = new QueryList<IgxColumnComponent>();
 
+    /**
+     * @hidden
+     */
     @ContentChild(IgxGridToolbarDirective, { read: TemplateRef })
-    public islandToolbarTemplate: TemplateRef<IgxGridToolbarTemplateContext>;
+    protected toolbarDirectiveTemplate: TemplateRef<IgxGridToolbarTemplateContext>;
 
+    /**
+     * @hidden
+     */
     @ContentChild(IgxPaginatorDirective, { read: TemplateRef })
-    public islandPaginatorTemplate: TemplateRef<any>;
+    protected paginatorDirectiveTemplate: TemplateRef<any>;
+
+    /**
+     * Gets the toolbar template for each child grid created from this row island.
+    */
+    @Input()
+    public get toolbarTemplate(): TemplateRef<IgxGridToolbarTemplateContext> {
+        return this._toolbarTemplate || this.toolbarDirectiveTemplate;
+    }
+
+    /**
+     * Sets the toolbar template for each child grid created from this row island.
+    */
+    public set toolbarTemplate(template: TemplateRef<IgxGridToolbarTemplateContext>) {
+        this._toolbarTemplate = template;
+    }
```

Comment:
No need for so much comments; just one will do. You can do `Get/Set` like others in the grid
```suggestion
    /**
     * Get/Set the toolbar template for each child grid created from this row island.
    */
    @Input()
    public get toolbarTemplate(): TemplateRef<IgxGridToolbarTemplateContext> {
        return this._toolbarTemplate || this.toolbarDirectiveTemplate;
    }

    public set toolbarTemplate(template: TemplateRef<IgxGridToolbarTemplateContext>) {
        this._toolbarTemplate = template;
    }
```
Or even just "Toolbar template for each child grid.." honestly the get/set part I hope is pretty self-explanatory